gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/jupiter/http/vertx/VertxHttpServerResponse.java

```diff
@@ -108,6 +111,19 @@ public Completable end() {
         );
     }
 
+    private Completable ensureWebSocketClosed() {
+        return Completable.defer(
+            () -> {
+                final WebSocket webSocket = serverRequest.webSocket();
+                if (!webSocket.closed()) {
+                    return webSocket.close(INTERNAL_SERVER_ERROR.code(), INTERNAL_SERVER_ERROR.reasonText());
```

Comment:
Must it necessary be a internal server error to arrive here ?

Reply:
I've set the equivalent error that is set by the failure processor (but for websocket).